### ISSUE TITLE
Document MotiVE paper reproduction, motivev2 intent, and prediction outputs

### ISSUE THREAD

**Issue by shntnu** [carpenter-singh-lab/2024_Arevalo_NeurIPS_MotiVE#10]
## Purpose

This issue is the canonical record of how the paper-era MotiVE workflow, John's post-publication `motivev2` work, and Shantanu's `motivev2` fork relate.
It documents what each repository line is for, what has been reproduced, and what prediction outputs currently exist.
It consolidates #11 so that branch intent, reproduction evidence, and prediction semantics remain in one place.

## Bottom line

MotiVE currently has two distinct uses.
The paper-era workflow supports comparison with the published results, while `motivev2` supports post-publication experimentation and candidate ranking.

| Repository line | Reference commit | Meaning |
|---|---|---|
| First release | `c7b7b9c8b5ce73e55c995869ec7287aea41f0b6a` | The version John identified as corresponding to the published results. |
| Upstream `main` | [`6e0f92cce3a38a73c9c7769bc21a29d40a1c0a1c`](https://github.com/carpenter-singh-lab/2024_Arevalo_NeurIPS_MotiVE/commit/6e0f92cce3a38a73c9c7769bc21a29d40a1c0a1c) | The current paper-era executable workflow used for the five-seed reproduction below. |
| Upstream `motivev2` | `2409ca5` | John's post-publication experimental continuation. |
| Shantanu's `motivev2` fork | `6452331f809991e2b910ded88a333b1907f23ed5` | A strict descendant of upstream `motivev2` with portability, environment, server, grid-running, fix, and documentation work. |

The retained `motivev2` outputs provide unvalidated rankings of candidate compound-ORF gene relationships within benchmark candidate sets.
They do not provide a finalized compound list for every ORF or CRISPR across JUMP.

## John's stated intent for the post-publication work

John directly described the post-publication work as adding other baselines, exploring ideas to improve the results, and adding more analysis.
He also said that these changes eventually broke reproducibility of the original version. [msg:19f56a88926393fd]

John separately said that he moved the GeneCompound matching repository to Snakemake "to make it more flexible" and to prepare for the next things the team wanted to try. [msg:C024LG7T4P7:1731719167.780699]
Related work tested newly combined profiles with the MotiVE baseline and compared them with the published results. [msg:C01AF25CQLT:1724268529.321649]

The substantive work in upstream `motivev2` was John Arevalo's work.
Future documentation should preserve that attribution.

### What remains inferred

The available messages describe the goals of the post-publication work without using the branch name `motivev2`.
The connection to that branch follows from context, timing, and correspondence with the commit history.

The implementation supports this interpretation.
Upstream `motivev2` added parameter exploration, BPR loss, mAP evaluation, all-pairs Cartesian inference, success-at-k metrics, random and KNN baselines, phenotypic-activity analysis, new profile support, and Snakemake orchestration.

The narrower interpretation that `motivev2` emphasizes ranking and candidate prioritization is also inferred from the implementation.
John's direct description remains the primary statement of intent.

## Paper-comparable reproduction

The paper-era workflow remains the appropriate path for reproducing the published experiment.
John identified the first release as the paper version and later described a successful five-run test as using `main`. [msg:19f56a88926393fd]

Upstream `main` had not changed between 2026-01-13 and John's July 2026 message, so his reference to `main` resolves to `6e0f92cce3a38a73c9c7769bc21a29d40a1c0a1c`.
His exact local working-tree state was not independently recorded.

### Independent five-seed demonstration

On 2026-08-24, I ran the GraphSAGE Cell Painting cold-target experiment on `spirit` from upstream `main` at `6e0f92c`.
The run used `configs/train/st_expanded/cold_target/gnn_cp.json`, ORF targets, the `st_expanded` graph, 1,000 epochs, and seeds `2024313` through `2024317`.

| Seed | Hits@500 | Precision@500 | F1 |
|---:|---:|---:|---:|
| 2024313 | 0.262677 | 0.838000 | 0.340202 |
| 2024314 | 0.227556 | 0.762000 | 0.350656 |
| 2024315 | 0.222776 | 0.732000 | 0.320780 |
| 2024316 | 0.227140 | 0.730000 | 0.336330 |
| 2024317 | 0.223400 | 0.746000 | 0.344956 |
| **Mean +/- sample SD** | **0.232710 +/- 0.016889** | **0.761600 +/- 0.044596** | **0.338585 +/- 0.011303** |
| John's July rerun | 0.2490 +/- 0.019 | 0.7868 +/- 0.018 | 0.3329 +/- 0.008 |
| Paper Table 4 | 0.1988 +/- 0.013 | 0.7056 +/- 0.031 | 0.3494 +/- 0.012 |

The new means fall within one combined standard deviation of John's rerun for all three metrics.
The result supports John's conclusion that this configuration produces results reasonably close to Table 4.

### Data and environment

The demonstration used the published S3 snapshot directly: 115 objects totaling 4,502,028,365 bytes.

```bash
aws s3 sync --no-sign-request \
  s3://cellpainting-gallery/cpg0034-arevalo-su-motive/broad/workspace/publication_data/2024_MOTIVE .
```

The run used Python 3.11.16, PyTorch 2.1.2 with CUDA 11.8, PyG 2.5.0, PyG Lib 0.4.0, and an NVIDIA H100 NVL.
The model and evaluation source were unchanged.
Each run set `train.SEED` and called `torch.manual_seed(seed)` before model construction.

The historical environment does not resolve cleanly for an H100 through Pixi's strict channel-priority solver.
The historical stack also creates hundreds of CPU threads per process, so the five runs should execute sequentially or with explicit thread limits.

This demonstration tested only the direct S3 snapshot.
Whether the Snakemake path produces byte-identical inputs remains unknown.

## Why `motivev2` is not a numerical paper reproduction

Shantanu revived `motivev2` and ran a broad benchmark grid from fork commit `6452331f809991e2b910ded88a333b1907f23ed5`.
The grid used `run_grid.sh`, shared hyperparameters from `gnn.json`, 200 training epochs, and the sampled and Cartesian evaluation workflows.

The grid recovered qualitative paper trends, including strong GNN performance and cold-target as the hardest split.
Its absolute values should not be presented as reproductions of Tables 3 and 4 because the workflow differs materially from the paper workflow:

- BPR loss replaces BCE loss.
- Minimum validation loss replaces validation Hits@500 for model selection.
- Shared GNN hyperparameters replace per-configuration tuning.
- Sampled and Cartesian evaluation modes extend the original evaluation.
- mAP and success-at-k extend the original metrics.

The defensible description is a modernized qualitative reproduction and post-publication experiment.

## Prediction outputs currently available

The retained sweep contains 72 `results.parquet` files from 36 configurations evaluated in sampled and Cartesian modes.
All retained results target ORFs; none target CRISPR perturbations.

The underlying `st_expanded/orf` data contain 3,632 compounds and 11,509 genes.
Each Cartesian result scores only compounds and genes represented in that configuration's test supervision set, rather than the full 3,632 x 11,509 cross-product.

For example, the random-split, `st_expanded`, GIN, Cell Painting Cartesian result contains 3,517,624 scored pairs across 1,562 compounds and 2,252 genes.
Of those pairs, 4,941 are held-out test annotations.

The `y_true` field identifies only held-out test annotations.
Train and validation annotations can appear as apparently negative pairs unless all known train, validation, and test annotations are excluded separately.

BPR scores support rankings within a run.
They are not calibrated probabilities.
Pairs absent from the available annotation tables should be called "unannotated," not "novel" or "validated."

### Minimal inventory and extraction

The attached [`summarize_predictions.py`](https://github.com/user-attachments/files/31400497/summarize_predictions.py) script inventories every retained result and extracts the top 20 unannotated compounds per gene from every Cartesian configuration.

Run it on `spirit` with:

```bash
cd /work/users/shsingh/GitHub/misc/2024_Arevalo_NeurIPS_MotiVE
CONDA_OVERRIDE_CUDA=12.0 pixi run python summarize_predictions.py
```

The script produces two files:

- `prediction_inventory.csv` contains 72 run-level rows with configurations, counts, metrics, and source paths.
- `prediction_top20_unannotated.parquet` contains 1,295,280 candidate rows after excluding all known train, validation, and test annotations.

Both generated files are available in [`motivev2-prediction-summaries.zip`](https://github.com/user-attachments/files/31400519/motivev2-prediction-summaries.zip).
The original attachment context remains in #11.

The generated copies on `spirit` are:

```text
/work/users/shsingh/GitHub/misc/2024_Arevalo_NeurIPS_MotiVE/prediction_inventory.csv
/work/users/shsingh/GitHub/misc/2024_Arevalo_NeurIPS_MotiVE/prediction_top20_unannotated.parquet
```

The inventory begins:

```text
target_type,split,graph,model,initialization,hash,infer_mode,pairs,compounds,genes,held_out_annotations,all_annotations,unannotated_pairs
orf,random,bipartite,bilinear,cp,97b583,cartesian,3517624,1562,2252,4941,17597,3500027
orf,random,bipartite,bilinear,cp,97b583,sampled,103761,2256,4479,4941,4941,98820
orf,random,bipartite,gin,cp,cf2f07,cartesian,3517624,1562,2252,4941,17597,3500027
orf,random,bipartite,gin,cp,cf2f07,sampled,499041,2280,4479,4941,4941,494100
```

The extracted rankings begin:

```text
target_type split  graph     model     init hash   compound_inchikey14 gene   score    rank
orf         random bipartite bilinear  cp   97b583 WPHKIQPVPYJNAX      AADAC  0.999988 1
orf         random bipartite bilinear  cp   97b583 KYRVNWMVYQXFEU      AADAC  0.999905 2
orf         random bipartite bilinear  cp   97b583 CSGQVNMSRKWUSH      AADAC  0.999876 3
orf         random bipartite bilinear  cp   97b583 IAKHMKGGTNLKSZ      AADAC  0.999772 4
orf         random bipartite bilinear  cp   97b583 MTJHLONVHHPNSI       AADAC  0.999710 5
```

## Supported wording

> From MotiVE, we have unvalidated model-generated rankings of candidate compound-ORF gene relationships within the benchmark candidate sets.
> We have not yet produced a finalized compound list for every ORF or CRISPR across JUMP.

## Remaining work

- [ ] Update the README to distinguish paper reproduction from `motivev2` experimentation.
- [ ] Record the clean-worktree status and complete command for John's July rerun if that provenance can still be recovered.
- [ ] Verify whether the direct S3 and Snakemake data paths produce identical inputs.
- [ ] Decide whether `summarize_predictions.py` and the generated summaries should live in the repository or a release rather than only as issue attachments and server files.
- [ ] Document explicitly that the retained rankings cover ORFs and benchmark candidate sets, not every ORF or CRISPR across JUMP.
- [ ] Treat prospective compound lists and experimental validation as future work if the project decides to pursue them.

**Comment by shntnu** [comment]
## Five-seed reproduction on Spirit

I reproduced the GraphSAGE Cell Painting cold-target experiment on `spirit` on 2026-08-24.

### Provenance

- Commit: [`6e0f92cce3a38a73c9c7769bc21a29d40a1c0a1c`](https://github.com/carpenter-singh-lab/2024_Arevalo_NeurIPS_MotiVE/commit/6e0f92cce3a38a73c9c7769bc21a29d40a1c0a1c)
- Config: `configs/train/st_expanded/cold_target/gnn_cp.json`
- Target type: ORF
- Graph: `st_expanded`
- Split: cold-target
- Epochs: 1,000
- Seeds: `2024313` through `2024317`
- Data: direct `aws s3 sync` from the published `2024_MOTIVE` prefix
- Published snapshot: 115 objects, 4,502,028,365 bytes
- GPU: NVIDIA H100 NVL
- Python: 3.11.16
- PyTorch: 2.1.2, CUDA 11.8
- PyG: 2.5.0
- PyG Lib: 0.4.0

Upstream `main` has not changed since 2026-01-13, so John's July 2026 reference to `main` resolves to the commit above.

I did not modify the model or evaluation source.
For each run, I set `train.SEED` and called `torch.manual_seed(seed)` before model construction so that initialization and sampling used the same recorded seed.

### Results

| Seed | Hits@500 | Precision@500 | F1 |
|---:|---:|---:|---:|
| 2024313 | 0.262677 | 0.838000 | 0.340202 |
| 2024314 | 0.227556 | 0.762000 | 0.350656 |
| 2024315 | 0.222776 | 0.732000 | 0.320780 |
| 2024316 | 0.227140 | 0.730000 | 0.336330 |
| 2024317 | 0.223400 | 0.746000 | 0.344956 |
| **Mean +/- sample SD** | **0.232710 +/- 0.016889** | **0.761600 +/- 0.044596** | **0.338585 +/- 0.011303** |
| John's July rerun | 0.2490 +/- 0.019 | 0.7868 +/- 0.018 | 0.3329 +/- 0.008 |
| Paper Table 4 | 0.1988 +/- 0.013 | 0.7056 +/- 0.031 | 0.3494 +/- 0.012 |

The new means are within one combined standard deviation of John's rerun for all three metrics.
They also support John's conclusion that this configuration produces results reasonably close to Table 4.

### Environment and execution notes

The historical `environment.yml` does not resolve cleanly for an H100 through Pixi's strict channel-priority solver.
A flexible solve initially selected a conda-forge PyTorch build without `sm_90` support.
I therefore selected the official PyTorch 2.1.2 CUDA 11.8 build and installed the matching PyG Lib wheel for PyTorch 2.1/CUDA 11.8.

The historical stack also creates hundreds of CPU threads per process on this server.
Five unbounded concurrent runs oversubscribed the machine, so I completed the runs sequentially after pausing the extras.
Future instructions should recommend sequential execution or explicit thread limits.

This run used the direct S3 snapshot only.
Whether the Snakemake path produces byte-identical inputs remains untested.

**Comment by shntnu** [comment]
Superseded by https://github.com/carpenter-singh-lab/2024_Arevalo_NeurIPS_MotiVE/issues/12. The new issue preserves the relevant facts, sources, results, and artifact links as a compact chronological record.
